modules/apps/29-fee/keeper/grpc_query.go

```diff
@@ -24,22 +27,30 @@ func (k Keeper) IncentivizedPackets(c context.Context, req *types.QueryIncentivi
 
 	ctx := sdk.UnwrapSDKContext(c).WithBlockHeight(int64(req.QueryHeight))
 
-	var packets []*types.IdentifiedPacketFee
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.FeeInEscrowPrefix))
-	_, err := query.Paginate(store, req.Pagination, func(_, value []byte) error {
-		result := k.MustUnmarshalFee(value)
-		packets = append(packets, &result)
+	var identifiedPackets []types.IdentifiedPacketFees
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.FeesInEscrowPrefix))
+	_, err := query.Paginate(store, req.Pagination, func(key, value []byte) error {
+		keySplit := strings.Split(string(key), "/")
+
+		channelID, portID := keySplit[2], keySplit[1]
+		seq, err := strconv.ParseUint(keySplit[3], 10, 64)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
Is this correct? From my investigation, pagination doesn't include the prefix iterated on in the return value. I guess there's a trailing "/" leading the key, which is subtle

So I think the key is something like "/transfer/channel-0/5"

Reply:
I'm kinda wondering if we should just reconstruct the full key and then use the parse functions in keys.go

Reply:
Yeah, correct, so a paginated query doesn't include the prefix, but because of the leading `"/"` when we split using the slash separator the first element is an empty string. So with your example, `keySplit` ends up as a 4 element slice: 
`[]string{"", "transfer", "channel-0", "5"}`.

I do think since you've added the parse function we should try to leverage it here. It should be something like below, right?
```
packetID, err := types.ParseKeyFeesInEscrow(types.FeesInEscrowPrefix + string(key))
```